addons/mail/static/src/backend_components/many2one_avatar_user/many2one_avatar_user.js

```diff
@@ -0,0 +1,96 @@
+/** @odoo-module **/
+
+import { registry } from "@web/core/registry";
+import { Many2OneField } from '@web/views/fields/many2one/many2one_field';
+import { useCommand } from "@web/core/commands/command_hook";
+
+const { Component } = owl;
+
+export class Many2oneAvatarUser extends Many2OneField {
+
+    setup() {
+        super.setup();
+        const provide = async (env, options) => {
+            const records = await this._searchAssignTo(options.searchValue, 10);
+            return records.map((record) => ({
+                name: record[1],
+                action: () => {
+                    this.update([{ id: record[0], display_name: record[1] }]);
+                }
+            }));
+        };
+        useCommand(
+            this.env._t('Assign to ...'),
+            () => {
+                return {
+                    configByNamespace: {
+                        default: {
+                            emptyMessage: this.env._t("No users found"),
+                        },
+                    },
+                    placeholder: this.env._t("Select a user..."),
+                    providers: [{ provide }],
+                };
+            },
+            {
+                category: "smart_action",
+                hotkey: "alt+i",
+                global: true
+            }
+        );
+        useCommand(
+            this.env._t("Assign/unassign to me"),
+            () => {
+                if (this.env.services.user.userId === this.props.value[0]) {
+                    this.update([{}]);
+                } else {
+                    this.update([{ id: this.env.services.user.userId, display_name: this.env.services.user.name }]);
+                }
+            },
+            {
+                category: "smart_action",
+                hotkey: "alt+shift+i",
+                global: true,
+            }
+        );
+    }
+
+    async _searchAssignTo(searchValue, limit) {
+        const value = searchValue.trim();
+
+        const nameSearch = await this.orm.call(
+            this.props.relation,
+            'name_search',
+            {},
+            {
+                name: value,
+                args: this.getDomain(),
+                operator: "ilike",
+                limit: limit + 1,
+                context: this.context,
+            }
+        );
+        return nameSearch;
+    }
```

Comment:
Looks like previous code had some "blacklist" in domain.
I'm trying to guess what's the functional motive... I think it's to not prompt assign someone that is the record itself? (e.g. do not prompt to assign "Marc Demo" when on the contact page of "Marc Demo")

I'm not sure... And if that was really the case, then it's not a big deal to have it prompt (and why not allow it?), and also the code was stupid because filtering just on id, so if by chance a project task has same id as "Marc Demo", looks like cannot prompt "Marc Demo" to assign to this task...

I can be all wrong. Honestly the previous code is too hard to understand. Just pointing out that some previous behaviour is missing here, but maybe it's actually an improvement!

Reply:
> I'm trying to guess what's the functional motive... I think it's to not prompt assign someone that is the record itself? (e.g. do not prompt to assign "Marc Demo" when on the contact page of "Marc Demo")

That is what I guessed too, but upon testing it, nope, it wasn't working that way.

Reply:
I think it's worth not putting it back. If it really breaks something, better make a proper `[FIX]` that tells exactly what it desired by this code, if it ever happens.